src/control/lib/netdetect/netdetect.go

```diff
@@ -537,6 +541,42 @@ func GetDeviceAlias(device string) (string, error) {
 	return C.GoString(node.name), nil
 }
 
+// mockGetDeviceAlias is a complete method to find an alias for the device name provided.
+// Because this function uses the system device list to rule out proper matching siblings,
+// a mock system devices list is provided so that the testing environment can simulate
+// the topology under test.
+func mockGetDeviceAlias(device string, mockSystemDevices []string) (string, error) {
```

Comment:
This is certainly an improvement and a step in the right direction in terms of improving the testability of this library. It's unfortunate that this function duplicates so much code from the real function, though.

Something to think about for a future iteration might be to create a Scanner (or something along those lines) struct which provides a place to store state and encapsulate functionality. It would expose public methods that call private methods which can be stubbed or mocked by the test setup. In the ideal scenario, your test cases would exercise as much of the Go code as possible without having any dependency on talking to libfabric. The hwloc dependency doesn't bother me as much because it provides a lot of value to testing with its ability to load mock configurations from xml.